include/deal.II/multigrid/mg_transfer_global_coarsening.templates.h

```diff
@@ -1649,6 +1649,36 @@ MGTwoLevelTransfer<dim, LinearAlgebra::distributed::Vector<Number>>::prolongate(
 
   for (const auto &scheme : schemes)
     {
+      // identity -> take short cut and work directly on global vectors
+      if (scheme.degree_fine == scheme.degree_coarse)
+        {
+          const unsigned int *indices_fine =
+            scheme.level_dof_indices_fine.data();
+          const unsigned int *indices_coarse =
+            scheme.level_dof_indices_coarse.data();
+          const Number *w = nullptr;
```

Comment:
Can we use slightly more verbose variable names?
```suggestion
          const Number *weights = nullptr;
```

Reply:
Done!